### PR TITLE
RA-1077 Configured modules and imports/exports for Webpack

### PR DESCRIPTION
**Pull request by rkorytkowski**
Picked up Pawel's work from
https://github.com/rkorytkowski/openmrs-owa-conceptdictionary/pull/29

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,205 @@
+'use strict';
```
Comment (psbrandt) [inline webpack.config.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/pascal_2/openmrs-owa-conceptdictionary/file/2307931000/issues/source?bid=3196969&fileBranchId=3196971#l1)

File: test/unit/conceptSearch.controller.spec.js [hunk 2]
```diff
@@ -71,9 +71,7 @@ describe('Concept dictionary controllers', function() {
 
 
             scope = $rootScope.$new();
-
-            ctrl = $controller('ConceptSearch', {$scope: scope, $location: location});
-
+            ctrl = $controller('ConceptSearchController', {$scope: scope, $location: location});
```
Comment (psbrandt) [inline test/unit/conceptSearch.controller.spec.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/pascal_2/openmrs-owa-conceptdictionary/file/2307930997/issues/source?bid=3196969&fileBranchId=3196971#l74)

File: webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+        chunkFilename: isProd ? '[name].[hash].js' : '[name].bundle.js'
+    };
+
+    /**
+     * Devtool
+     * Reference: http://webpack.github.io/docs/configuration.html#devtool
+     * Type of sourcemap to use per build type
+     */
+    if (isTest) {
+        config.devtool = 'inline-source-map';
```
Comment (psbrandt) [inline webpack.config.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/pascal_2/openmrs-owa-conceptdictionary/file/2307931000/issues/source?bid=3196969&fileBranchId=3196971#l64)

File: webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+     */
+    config.plugins = [];
+
+    // Skip rendering index.html in test mode
+    if (!isTest) {
+        // Reference: https://github.com/ampedandwired/html-webpack-plugin
+        // Render index.html
+        config.plugins.push(
+            new HtmlWebpackPlugin({
+                template: './app/admin.html',
```
Comment (psbrandt) [inline webpack.config.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/pascal_2/openmrs-owa-conceptdictionary/file/2307931000/issues/source?bid=3196969&fileBranchId=3196971#l160)

File: webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+    // Skips node_modules and files that end with .test.js
+    if (isTest) {
+        config.module.preLoaders.push({
+            test: /\.js$/,
+            exclude: [
+                /node_modules/,
+                /\.spec\.js$/
+            ],
+            loader: 'isparta-instrumenter'
+        })
```
Comment (psbrandt) [inline webpack.config.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/pascal_2/openmrs-owa-conceptdictionary/file/2307931000/issues/source?bid=3196969&fileBranchId=3196971#l133)

File: webpack.config.js [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+var CopyWebpackPlugin = require('copy-webpack-plugin');
+/**
+ * Env
+ * Get npm lifecycle event to identify the environment
+ */
+var ENV = process.env.npm_lifecycle_event;
+var isTest = ENV === 'test' || ENV === 'test-watch';
+var isProd = false;
+
+module.exports = function makeWebpackConfig () {
```
Comment (psbrandt) [inline webpack.config.js]: Codacy Issue found: [Wrap an immediate function invocation in parentheses.](https://www.codacy.com/app/pascal_2/openmrs-owa-conceptdictionary/file/2307931000/issues/source?bid=3196969&fileBranchId=3196971#l17)

File: webpack.config.js [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+            new webpack.optimize.DedupePlugin(),
+
+            // Reference: http://webpack.github.io/docs/list-of-plugins.html#uglifyjsplugin
+            // Minify all javascript, switch loaders to minimizing mode
+            //new webpack.optimize.UglifyJsPlugin(),
+
+            // Copy assets from the public folder
+            // Reference: https://github.com/kevlened/copy-webpack-plugin
+            new CopyWebpackPlugin([{
+                from: __dirname + '/app'
```
Comment (psbrandt) [inline webpack.config.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/pascal_2/openmrs-owa-conceptdictionary/file/2307931000/issues/source?bid=3196969&fileBranchId=3196971#l189)

File: webpack.config.js [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+        config.plugins.push(
+            new HtmlWebpackPlugin({
+                template: './app/admin.html',
+                inject: 'body'
+            }),
+
+            // Reference: https://github.com/webpack/extract-text-webpack-plugin
+            // Extract css files
+            // Disabled when in test mode or not in build mode
+            new ExtractTextPlugin('[name].[hash].css', {disable: !isProd})
```
Comment (psbrandt) [inline webpack.config.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/pascal_2/openmrs-owa-conceptdictionary/file/2307931000/issues/source?bid=3196969&fileBranchId=3196971#l167)

File: webpack.config.js [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+        // Initialize module
+    config.module = {
+        preLoaders: [],
+        loaders: [{
+            // JS LOADER
+            // Reference: https://github.com/babel/babel-loader
+            // Transpile .js files using babel-loader
+            // Compiles ES6 and ES7 into ES5 code
+            test: /\.js$/,
+            loader: 'babel',
```
Comment (psbrandt) [inline webpack.config.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/pascal_2/openmrs-owa-conceptdictionary/file/2307931000/issues/source?bid=3196969&fileBranchId=3196971#l87)

File: webpack.config.js [hunk 10]
```diff
@@ -0,0 +1,205 @@
...
+    // Instrument JS files with Isparta for subsequent code coverage reporting
+    // Skips node_modules and files that end with .test.js
+    if (isTest) {
+        config.module.preLoaders.push({
+            test: /\.js$/,
+            exclude: [
+                /node_modules/,
+                /\.spec\.js$/
+            ],
+            loader: 'isparta-instrumenter'
```
Comment (psbrandt) [inline webpack.config.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/pascal_2/openmrs-owa-conceptdictionary/file/2307931000/issues/source?bid=3196969&fileBranchId=3196971#l132)

File: webpack.config.js (hunk 1 above)
Comment (psbrandt) [inline webpack.config.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/pascal_2/openmrs-owa-conceptdictionary/file/2307931000/issues/source?bid=3196969&fileBranchId=3196971#l1)

File: test/unit/conceptSearch.controller.spec.js (hunk 2 above)
Comment (psbrandt) [inline test/unit/conceptSearch.controller.spec.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/pascal_2/openmrs-owa-conceptdictionary/file/2307930997/issues/source?bid=3196969&fileBranchId=3196971#l74)

File: webpack.config.js (hunk 3 above)
Comment (psbrandt) [inline webpack.config.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/pascal_2/openmrs-owa-conceptdictionary/file/2307931000/issues/source?bid=3196969&fileBranchId=3196971#l64)

File: webpack.config.js (hunk 4 above)
Comment (psbrandt) [inline webpack.config.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/pascal_2/openmrs-owa-conceptdictionary/file/2307931000/issues/source?bid=3196969&fileBranchId=3196971#l160)

File: webpack.config.js (hunk 5 above)
Comment (psbrandt) [inline webpack.config.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/pascal_2/openmrs-owa-conceptdictionary/file/2307931000/issues/source?bid=3196969&fileBranchId=3196971#l133)

File: webpack.config.js (hunk 6 above)
Comment (psbrandt) [inline webpack.config.js]: Codacy Issue found: [Wrap an immediate function invocation in parentheses.](https://www.codacy.com/app/pascal_2/openmrs-owa-conceptdictionary/file/2307931000/issues/source?bid=3196969&fileBranchId=3196971#l17)

File: webpack.config.js (hunk 7 above)
Comment (psbrandt) [inline webpack.config.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/pascal_2/openmrs-owa-conceptdictionary/file/2307931000/issues/source?bid=3196969&fileBranchId=3196971#l189)

File: webpack.config.js (hunk 8 above)
Comment (psbrandt) [inline webpack.config.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/pascal_2/openmrs-owa-conceptdictionary/file/2307931000/issues/source?bid=3196969&fileBranchId=3196971#l167)

File: webpack.config.js (hunk 10 above)
Comment (psbrandt) [inline webpack.config.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/pascal_2/openmrs-owa-conceptdictionary/file/2307931000/issues/source?bid=3196969&fileBranchId=3196971#l132)

File: webpack.config.js (hunk 9 above)
Comment (psbrandt) [inline webpack.config.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/pascal_2/openmrs-owa-conceptdictionary/file/2307931000/issues/source?bid=3196969&fileBranchId=3196971#l87)

**Comment by rkorytkowski** [comment]
Closing for now... It needs to be reworked as code has been restructured.

**Pull request closed without merging**